torch/_dynamo/trace_rules.py

```diff
@@ -3750,12 +3753,46 @@ def lookup(obj):
     return lookup_inner(obj)
 
 
+# also takes config.dont_skip_tracing into account
 def lookup_inner(
     obj,
     name=None,
     filename=None,
     is_direct_call=True,
     reasons: Union[None, set[str]] = None,
+):
+    result = _lookup_inner(
+        obj,
+        name=name,
+        filename=filename,
+        is_direct_call=is_direct_call,
+        reasons=reasons,
+    )
+    # there are still some modules we should absolutely NOT trace into - e.g. torch._dynamo, as this
+    # can result in really weird tracing behaviors.
+    if config.dont_skip_tracing and result is SkipFunctionVariable:
```

Comment:
For my understanding: if we did mark something as traceable in torch._dynamo, it would have returned e.g. UserFunctionVariable, and we would not get into this branch?

Reply:
Yes.